pymc/variational/callbacks.py

```diff
@@ -93,6 +93,77 @@ def flatten_shared(shared_list):
         return np.concatenate([sh.get_value().flatten() for sh in shared_list])
 
 
+class ReduceLROnPlateau(Callback):
+    """Reduce learning rate when the loss has stopped improving.
+
+    This is inspired by Keras' homonymous callback:
+    https://github.com/keras-team/keras/blob/v2.14.0/keras/callbacks.py
+
+    Parameters
+    ----------
+    optimizer: callable
+        PyMC optimizer
+    factor: float
+        factor by which the learning rate will be reduced: `new_lr = lr * factor`
+    patience: int
+        number of epochs with no improvement after which learning rate will be reduced
+    min_lr: float
+        lower bound on the learning rate
+    cooldown: int
+        number of iterations to wait before resuming normal operation after lr has been reduced
+    verbose: bool
+        false: quiet, true: update messages
+    """
+
+    def __init__(
+        self,
+        optimizer,
```

Comment:
Does the user have to provide this? Can it instead be inferred somehow from the host VI object? It's ugly to have to pass the optimizer twice (once for the VI itself, then again in the callback)

Reply:
Well, this would be great, but I haven't figured out whether it's possible. Probably one for someone more familiar with the codebase :)